diff --git a/pupil_src/shared_modules/calibration_routines/optimization_calibration/setup.py b/pupil_src/shared_modules/calibration_routines/optimization_calibration/setup.py
index 9ddf182e..e422d5c4 100644
--- a/pupil_src/shared_modules/calibration_routines/optimization_calibration/setup.py
+++ b/pupil_src/shared_modules/calibration_routines/optimization_calibration/setup.py
@@ -74,13 +74,8 @@ else:
                 opencv_libraries = [lib + '3' for lib in opencv_libraries]
                 break
     include_dirs = [np.get_include(), '/usr/local/include/eigen3', '/usr/include/eigen3', shared_cpp_include_path ,singleeyefitter_include_path, opencv_include_dir]
-    if platform.system() == 'Linux':
-        python_version = sys.version_info
-        # boost_python-py34
-        boost_lib = 'boost_python-py'+str(python_version[0])+str(python_version[1])
-    else:
-        boost_lib = 'boost_python3'
-    libs = ['ceres', boost_lib]+opencv_libraries
+    boost_lib = 'boost_python3'
+    libs = ['ceres', 'cholmod', 'cxsparse', 'glog', 'gomp', boost_lib]+opencv_libraries
     xtra_obj = []
     library_dirs = [opencv_library_dir]
 
diff --git a/pupil_src/shared_modules/pupil_detectors/setup.py b/pupil_src/shared_modules/pupil_detectors/setup.py
index 05c65d2f..cb3ecaa2 100644
--- a/pupil_src/shared_modules/pupil_detectors/setup.py
+++ b/pupil_src/shared_modules/pupil_detectors/setup.py
@@ -74,13 +74,8 @@ else:
                 opencv_libraries = [lib + '3' for lib in opencv_libraries]
                 break
     include_dirs = [np.get_include(), '/usr/local/include/eigen3','/usr/include/eigen3', shared_cpp_include_path, singleeyefitter_include_path, opencv_include_dir]
-    if platform.system() == 'Linux':
-        python_version = sys.version_info
-        # boost_python-py34
-        boost_lib = 'boost_python-py'+str(python_version[0])+str(python_version[1])
-    else:
-        boost_lib = 'boost_python3'
-    libs = ['ceres', boost_lib]+opencv_libraries
+    boost_lib = 'boost_python3'
+    libs = ['ceres', 'cholmod', 'cxsparse', 'glog', 'gomp', boost_lib]+opencv_libraries
     xtra_obj2d = []
     library_dirs = [opencv_library_dir]
 
